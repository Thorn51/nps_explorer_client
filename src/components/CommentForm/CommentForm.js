import React, { useContext } from "react";
import GlobalState from "../../context/GlobalState";

export default function CommentForm(props) {
  return (
    <form className="comment_form">
      <p>You are required to be logged in to submit comments.</p>
      <hr />
      <label htmlFor="comment">Leave your thoughts for {props.parkName}</label>
      <textarea
        name="comment"
        cols="30"
        rows="15"
        className="comment"
        id="comment"
      ></textarea>
      <button type="submit" className="form_button">
        Submit
      </button>
    </form>
  );
}
