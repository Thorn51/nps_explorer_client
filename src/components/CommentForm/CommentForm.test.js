import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import CommentForm from "./CommentForm";

describe("<CommentForm />", () => {
  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<CommentForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Renders the UI as expected", () => {
    const tree = renderer.create(<CommentForm />).toJSON();
    expect("tree").toMatchSnapshot();
  });
});
