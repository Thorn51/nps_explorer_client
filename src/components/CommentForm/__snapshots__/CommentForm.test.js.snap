// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentForm /> Renders the UI as expected 1`] = `"tree"`;
